input-profile.js

```diff
@@ -0,0 +1,249 @@
+import {Component, Property} from '@wonderlandengine/api';
+import {vec3, quat} from 'gl-matrix';
+const _TempVec = vec3.create();
+const _TempQuat = quat.create();
+
+export class InputProfile extends Component {
+    static TypeName = 'input-profile';
+    static Properties = {
+        controller: Property.object(),
+        handednessIndex: Property.enum(['left', 'right'], 0),
+        path: Property.string(
+            'https://cdn.jsdelivr.net/npm/@webxr-input-profiles/assets@1.0.13/dist/profiles/'
+        ),
+        customProfileFolder: Property.string(''),
+        preSpawnedController: Property.object(),
+    };
+
+    init() {
+        this._gamepadObjects = {};
+        this._modelLoaded = false;
+        this.controllerDeactivateFlag = false;
+        this.ProfileJSON = {};
+    }
+
+    start() {
+        if (this.engine.XRSession) {
+            this.engine.XRSession.addEventListener(
+                'inputsourceschange',
+                this.onInputSourcesChange.bind(this)
+            );
+        } else {
+            this.engine.onXRSessionStart.add((session) => {
+                session.addEventListener(
```

Comment:
th listener is only needed to be added once , 
since the component is being activated and deactivated over multiple times (by vr mode active switch) ,Moving it to onActivate gives undesired result due to multiple addition of event listeners 
We could add a listener on OnDeactivate though

Reply:
Then you should probably unregister it in `onDestroy`. I am not a fan of `vr-active-switch` because if you now want to disable the input-profile component, I guess you can't? Because the state will be changed by `vr-active-switch`.

In any case, leaking listeners shouldn't be allowed